ibis/backends/tests/test_client.py

```diff
@@ -202,3 +203,79 @@ def test_separate_database(con, alternate_current_database, current_data_db):
     db = con.database(current_data_db)
     assert db.name == current_data_db
     assert tmp_db.name == alternate_current_database
+
+
+@pytest.mark.only_on_backends(
+    ['sqlite', 'postgres', 'mysql'],
+    reason="run only if backend is SQLAlchemy based",
+)
+def test_insert_from_dataframe(con):
+
+    sch = ibis.schema(
+        [
+            ('first_name', 'string'),
+            ('last_name', 'string'),
+            ('department_name', 'string'),
+            ('salary', 'float64'),
+        ]
+    )
+
+    temp_table = 'temp_to_table'
+    con.drop_table(temp_table, force=True)
+    con.create_table(temp_table, schema=sch)
+    temporary = con.table(temp_table)
+    assert len(temporary.execute()) == 0
+
+    records = pd.DataFrame(
+        {
+            'first_name': ['X', 'Y', 'Z'],
+            'last_name': ['A', 'B', 'C'],
+            'department_name': ['XX', 'YY', 'ZZ'],
+            'salary': [400.0, 500.0, 600.0],
+        }
+    )
+
+    con.insert('temp_to_table', data_obj=records)
+    assert len(temporary.execute()) == 3
+    tm.assert_frame_equal(temporary.execute(), records)
```

Comment:
In both cases we should add tests for at least the different values of `if_exists`. Otherwise we can't know if your implementation is working. Or if future changes break it.

Reply:
Done. Added separate tests for each of the values of `if_exists` for both the cases. Please take a look. Thanks.